client/tasks/fills/tax/index.tsx

```diff
@@ -40,64 +32,22 @@ const Tax = ( { onComplete, query } ) => {
 	const { updateAndPersistSettingsForGroup } = useDispatch(
 		SETTINGS_STORE_NAME
 	);
-	const {
-		generalSettings,
-		isJetpackConnected,
-		isResolving,
-		pluginsToActivate,
-		tasksStatus,
-		taxSettings,
-	} = useSelect( ( select ) => {
-		const { getSettings } = select(
-			SETTINGS_STORE_NAME
-		) as SettingsSelector;
-		const { getActivePlugins } = select( PLUGINS_STORE_NAME );
-		const activePlugins = getActivePlugins();
-
-		return {
-			generalSettings: getSettings( 'general' ).general,
-			isJetpackConnected: select(
-				PLUGINS_STORE_NAME
-			).isJetpackConnected(),
-			isResolving:
-				! select( PLUGINS_STORE_NAME ).hasFinishedResolution(
-					'isJetpackConnected'
-				) ||
-				! select(
-					SETTINGS_STORE_NAME
-				).hasFinishedResolution( 'getSettings', [ 'general' ] ) ||
-				! select( ONBOARDING_STORE_NAME ).hasFinishedResolution(
-					'getTasksStatus'
-				),
-			pluginsToActivate: difference( AUTOMATION_PLUGINS, activePlugins ),
-			// @Todo this should be removed as soon as https://github.com/woocommerce/woocommerce-admin/pull/7841 is merged.
-			tasksStatus: select( ONBOARDING_STORE_NAME ).getTasksStatus(),
-			taxSettings: getSettings( 'tax' ).tax || {},
-		};
-	} );
-
-	const supportsAutomatedTaxes = () => {
-		const {
-			automatedTaxSupportedCountries = [],
-			taxJarActivated,
-		} = tasksStatus;
-
-		return (
-			! taxJarActivated && // WCS integration doesn't work with the official TaxJar plugin.
-			automatedTaxSupportedCountries.includes(
-				getCountryCode( generalSettings?.woocommerce_default_country )
-			)
-		);
-	};
-
-	const canAutomateTaxes = () => {
-		return (
-			hasCompleteAddress( generalSettings ) &&
-			! pluginsToActivate.length &&
-			isJetpackConnected &&
-			supportsAutomatedTaxes()
-		);
-	};
+	const { generalSettings, isResolving, taxSettings } = useSelect(
+		( select ) => {
+			const { getSettings, hasFinishedResolution } = select(
+				SETTINGS_STORE_NAME
+			) as SettingsSelector;
+
+			return {
+				generalSettings: getSettings( 'general' ).general,
+				isResolving: ! hasFinishedResolution( 'getSettings', [
+					'general',
+				] ),
+				// @Todo this should be removed as soon as https://github.com/woocommerce/woocommerce-admin/pull/7841 is merged.
+				taxSettings: getSettings( 'tax' ).tax || {},
+			};
+		}
+	);
 
 	const onManual = async () => {
```

Comment:
You  might wrap this, as well as `onAutomate()` and `onDisable()` in `useCallback`, as any components that are fed these as props will currently always rerender since you're instantiating new functions every time.

Reply:
Good call, @joelclimbsthings!